relay-quotas/src/cache.rs

```diff
@@ -117,9 +119,17 @@ where
                 return CachedQuota::new_needs_sync(total_local_use);
             }
 
-            let remaining = quota.limit - consumed;
-            let max_allowed_spend = usize::try_from(remaining).unwrap_or(usize::MAX)
-                * PERCENT_PRECISION
+            let remaining = usize::try_from(quota.limit - consumed).unwrap_or(usize::MAX);
+            let max_allowed_spend = remaining
+                // Normalize the remaining quota with the window size, to apply the ratio/divisor to the
+                // per second rate.
+                //
+                // This means we get a consistent behaviour for short (10s) quotas (e.g. abuse) as well
+                // as long (1h) quotas (e.g. spike protection) with a more predictable error.
+                / usize::try_from(quota.window).unwrap_or(1).max(1)
+                // Apply ratio precision, which is already pre-multiplied into `max_over_spend_divisor`.
+                * RATIO_PRECISION
+                // Apply the actual ratio with the pre-computed divisor.
                 / self.max_over_spend_divisor.get();
```

Comment:
Not new in this PR, but what's the advantage of inverting the ratio in the constructor and then inverting the divisor to get a ratio again here?